plugins/org.eclipse.sirius/src/org/eclipse/sirius/business/internal/image/ImageManagerForWorkspaceResource.java

```diff
@@ -111,6 +114,34 @@ public CreateImageFileProvider getCreateFileFunc(EObject contextObject, String s
         return new CreateImageFileProvider(createFileFunction, filename);
     }
 
+    /**
+     * Get the name of the existing images folder name if it exists; {@link #IMAGE_FOLDER_NAME} otherwise.
+     * 
+     * @param airdFile
+     *            the file used to retrieve an existing images folder in the project
+     * @return the name of the existing images folder name if it exists; {@link #IMAGE_FOLDER_NAME} otherwise.
+     */
+    private String getImagesFolderName(IResource airdFile) {
+        String imageFolderName = ImageManager.IMAGE_FOLDER_NAME;
+        IProject project = airdFile.getProject();
+        Optional<IFolder> optImageFolder = Optional.empty();
+        try {
+            optImageFolder = Arrays.stream(project.members()) //
+                    .filter(IFolder.class::isInstance) //
+                    .map(IFolder.class::cast) //
+                    .filter(folder -> ImageManager.IMAGE_FOLDER_NAME.equals(folder.getName().toLowerCase())) //
+                    .findFirst();
+        } catch (CoreException e) {
+
+        }
```

Comment:
Ok but I think `optImageFolder = Optional.empty()` inside the `catch` instead of before `try` would be easier to read.

Reply:
done